File: app/src/main/java/it/droidcon/testingdaggerrxjava/dagger/UserInteractorModule.java
package it.droidcon.testingdaggerrxjava.dagger;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;
import it.droidcon.testingdaggerrxjava.core.UserInteractor;
import it.droidcon.testingdaggerrxjava.core.gson.StackOverflowService;

@Module
public class UserInteractorModule {

    @Provides @Singleton
    public UserInteractor provideUserInteractor(StackOverflowService stackOverflowService) {
        return new UserInteractor(stackOverflowService);
    }
}

File: app/src/main/java/it/droidcon/testingdaggerrxjava/dagger/UserInteractorModule.kt
package it.droidcon.testingdaggerrxjava.dagger

import dagger.Module
import dagger.Provides
import it.droidcon.testingdaggerrxjava.core.UserInteractor
import it.droidcon.testingdaggerrxjava.core.gson.StackOverflowService
import javax.inject.Singleton

@Module open class UserInteractorModule {

    @Provides @Singleton open fun provideUserInteractor(stackOverflowService: StackOverflowService) =
            UserInteractor(stackOverflowService)
}
